extensions/resteasy-reactive/quarkus-resteasy-reactive/deployment/src/main/java/io/quarkus/resteasy/reactive/server/deployment/QuarkusServerEndpointIndexer.java

```diff
@@ -169,7 +169,7 @@ private void warnAboutMissingJsonProviderIfNeeded(ServerResourceMethod method, M
         if (!capabilities.isCapabilityWithPrefixMissing("io.quarkus.resteasy.reactive.json")) {
             return;
         }
-        if (hasJson(method) || isDefaultJson()) {
+        if (hasJson(method) || (hasNoTypesDefined(method) && isDefaultJson())) {
```

Comment:
@geoand could you have a closer look at this change?

Even when I defined a proper `APPLICATION_XML` content type, I had the warning because `isDefaultJson()` was true and we didn't test that other content types were defined.

Reply:
Seems reasonable